x-pack/libbeat/management/fleet/manager.go

```diff
@@ -129,28 +122,49 @@ func (cm *Manager) CheckRawConfig(cfg *common.Config) error {
 	return nil
 }
 
-func (cm *Manager) worker() {
-	defer cm.wg.Done()
+func (cm *Manager) OnConfig(s string) {
+	cm.client.Status(proto.StateObserved_CONFIGURING, "Updating configuration")
 
-	// Start worker loop: fetch + apply  new settings
-WORKERLOOP:
-	for {
-		select {
-		case cfg := <-cm.configChan:
-			blocks, err := cm.toConfigBlocks(cfg)
-			if err != nil {
-				cm.logger.Errorf("Could not apply the configuration, error: %+v", err)
-				continue WORKERLOOP
-			}
+	var configMap common.MapStr
+	uconfig, err := common.NewConfigFrom(s)
+	if err != nil {
+		err = errors.Wrap(err, "config blocks unsuccessfully generated")
+		cm.logger.Error(err)
+		cm.client.Status(proto.StateObserved_FAILED, err.Error())
+	}
 
-			if errs := cm.apply(blocks); !errs.IsEmpty() {
-				cm.logger.Errorf("Could not apply the configuration, error: %+v", errs)
-				continue WORKERLOOP
-			}
-		case <-cm.done:
-			return
-		}
+	err = uconfig.Unpack(&configMap)
+	if err != nil {
+		err = errors.Wrap(err, "config blocks unsuccessfully generated")
+		cm.logger.Error(err)
+		cm.client.Status(proto.StateObserved_FAILED, err.Error())
```

Comment:
does it make sense to continue if this failed?

Reply:
missed the `return` in all the branches here, fixed

marking `Failed` will result in Agent being notified and then it restarting the beat